src/VirtoCommerce.MarketingModule.Data/Services/EvaluationPolicies/BestRewardPromotionPolicy.cs

```diff
@@ -59,6 +59,19 @@ public async Task<PromotionResult> EvaluatePromotionAsync(IEvaluationContext con
                     }
                 }
 
+                //best payment promotion
```

Comment:
Seems this code is wrong as well as for the shipment method. if promoContext.PaymentMethodCode == null, reward still applied. Not sure it is correct.
Need to fix the code and write proper tests for both shipment and payment reward cases.

Reply:
actually, rewards are applied in Storefront and check conditions(PaymentMethodCode etc), look at https://github.com/VirtoCommerce/vc-storefront/blob/master/VirtoCommerce.Storefront.Model/Cart/Payment.cs#L145